sub-process.c

```diff
@@ -78,7 +78,12 @@ int subprocess_start(struct hashmap *hashmap, struct subprocess_entry *entry, co
 	int err;
 	struct child_process *process;
 
-	entry->cmd = cmd;
+	// BUGBUG most callers to subprocess_start() pass in "cmd" the value
```

Comment:
regarding your TODO in the commit message: we will not be able to drop the read_object_hook() from the Git code as we hope to unify the vfs- and features/sparse-checkout- branches.